src/widgets/Tabs/VaccineSelect.js

```diff
@@ -147,19 +165,65 @@ const VaccineSelectComponent = ({
 
       <FlexRow flex={1} alignItems="flex-end" justifyContent="flex-end">
         <PageButtonWithOnePress text={buttonStrings.cancel} onPress={onCancelPrescription} />
-        <PageButtonWithOnePress
-          text={buttonStrings.confirm}
-          style={{ marginLeft: 'auto' }}
-          isDisabled={!selectedBatches && !hasRefused}
-          onPress={confirmPrescription}
+
+        {wasVaccinatedToday() ? (
+          <PageButton
+            text={buttonStrings.confirm}
+            style={{ marginLeft: 'auto' }}
+            isDisabled={!selectedBatches && !hasRefused}
+            onPress={toggleConfirmDoubleDoseModal}
+          />
+        ) : (
+          <PageButtonWithOnePress
+            text={buttonStrings.confirm}
+            style={{ marginLeft: 'auto' }}
+            isDisabled={!selectedBatches && !hasRefused}
+            onPress={confirmPrescription}
+          />
+        )}
+
+        {wasVaccinatedToday() ? (
+          <PageButton
+            text={generalStrings.ok_and_next}
+            style={{ marginLeft: 5 }}
+            isDisabled={!selectedBatches && !hasRefused}
+            onPress={toggleConfirmAndRepeatDoubleDoseModal}
+          />
+        ) : (
+          <PageButtonWithOnePress
+            text={generalStrings.ok_and_next}
+            style={{ marginLeft: 5 }}
+            isDisabled={!selectedBatches && !hasRefused}
+            onPress={confirmAndRepeatPrescription}
+          />
+        )}
+      </FlexRow>
+      <PaperModalContainer
+        isVisible={confirmDoubleDoseModalOpen}
+        onClose={toggleConfirmDoubleDoseModal}
+      >
+        <PaperConfirmModal
+          questionText={modalStrings.confirm_double_dose}
+          confirmText={modalStrings.confirm}
+          cancelText={modalStrings.cancel}
+          onConfirm={confirmPrescription}
+          onCancel={toggleConfirmDoubleDoseModal}
+          withOnePress={true}
         />
-        <PageButtonWithOnePress
-          text={generalStrings.ok_and_next}
-          style={{ marginLeft: 5 }}
-          isDisabled={!selectedBatches && !hasRefused}
-          onPress={confirmAndRepeatPrescription}
+      </PaperModalContainer>
+      <PaperModalContainer
+        isVisible={confirmAndRepeatDoubleDoseModalOpen}
+        onClose={toggleConfirmAndRepeatDoubleDoseModal}
+      >
+        <PaperConfirmModal
+          questionText={modalStrings.confirm_double_dose}
+          confirmText={modalStrings.confirm}
+          cancelText={modalStrings.cancel}
+          onConfirm={confirmAndRepeatPrescription}
+          onCancel={toggleConfirmAndRepeatDoubleDoseModal}
+          withOnePress={true}
```

Comment:
Could we do something like:

```
const modalOnConfirm = confirmDoubleDoseModalOpen ? confirmPrescription : confirmRepeatPrescription
const modalOnCancel = confirmDoubleDoseModalOpen ? toggleConfirmDoubleDoseModal : toggleConfirmAndRepeatDoubleDoseModal
...
```

and use the same modal?

Also just noticed the `PageButton` has a `debounceTimer` prop

Reply:
> ```
> const modalOnConfirm = confirmDoubleDoseModalOpen ? confirmPrescription : confirmRepeatPrescription
> const modalOnCancel = confirmDoubleDoseModalOpen ? toggleConfirmDoubleDoseModal : toggleConfirmAndRepeatDoubleDoseModal
> ...
> ```
> 
> and use the same modal?

I'm not quite following sorry 🤣  - if the modal is open it shouldn't necessarily impact which function to execute I think? It just means they have either clicked `Confirm` or `OK & Next` but we don't know which - I was thinking I could add something to the PageButton onPress to store which one was last clicked, but not sure how 🤔

Reply:
It's possibly a terrible idea, happy for you to say "no ew". It's more using which modal is open rather than if the modal is open to determine the callback.

```
const modalOnClose = confirmDoubleDoseModalOpen
  ? toggleConfirmDoubleDoseModal
  : toggleConfirmAndRepeatDoubleDoseModal;

const modalOnConfirm = confirmDoubleDoseModalOpen
  ? confirmPrescription
  : confirmAndRepeatPrescription;

const modalOpen = confirmDoubleDoseModalOpen || confirmAndRepeatDoubleDoseModalOpen;

const Modal = () => (
  <PaperModalContainer isVisible={modalOpen} onClose={modalOnClose}>
    <PaperConfirmModal
      questionText={modalStrings.confirm_double_dose}
      confirmText={modalStrings.confirm}
      cancelText={modalStrings.cancel}
      onConfirm={modalOnConfirm}
      onCancel={modalOnClose}
    />
  </PaperModalContainer>
);
```

I think this works..?

Reply:
Ohhhh thanks I see, so still maintaining two toggle states for the different modals but using only one modal component.
Haha okay. Maybe a bit ew but I think a lot less ew than my version 😹

Reply:
Works nicely :) - yoink 57b1f036 (and 3a847908 for consistency)